### PR TITLE
Renaming & test case added

### PR DESCRIPTION
**Pull request by zypeh**

### CODE REVIEW

File: src/node.rs [hunk 1]
```diff
@@ -1,40 +1,42 @@
...
+    pub fn cons(node1: T, node2: T) -> Pair<T> {
+        Pair(Box::new(node1), Box::new(node2))
     }
-}
 
-pub fn cdr(node: Node) -> Result<Node, &'static str> {
-    match node {
-        Node::Pair(_, b) => Result::Ok(*b),
-        _ => Result::Err("Cannot apply cdr to Pair")
+    pub fn car(pair: Pair<T>) -> T {
```
Comment (hch12907) [inline src/node.rs]:
```suggestion
    pub fn car(self) -> T {
```

File: src/node.rs [hunk 2]
```diff
@@ -1,40 +1,42 @@
...
+    TypeNode(Type),
+    Unit,
 }
 
 #[derive(Clone, PartialEq, Debug)]
 pub enum Type {
-    Universe(i64),
-    Atom,
-    Pair(Box<Type>, Box<Type>)
+    Universe(u32),
```
Comment (hch12907) [inline src/node.rs]: `u64`? It was `i64`.

File: src/lib.rs [hunk 3]
```diff
@@ -2,68 +2,125 @@ pub mod node;
...
     }
 
     #[test]
-    fn list_car() {
-        use crate::node::car;
+    fn pair_car() {
         assert_eq!(
-            car(vec![Atom("ratatouille"), Atom("baguette")]),
+            Pair::car(Pair {
+                0: Box::new(Atom("ratatouille")),
```
Comment (hch12907) [inline src/lib.rs]: It's more common to see
```rust
Pair (
    Box::new(Atom("...")),
    Box::new(Atom("..."))
)
```

Comment (hch12907) [reply src/lib.rs]: Ditto for the others

File: src/node.rs [hunk 4]
```diff
@@ -1,40 +1,42 @@
...
+    AtomTy,
+    PairTy(Pair<Type>),
+    Unit,
 }
 
-pub fn cons(node: Node, atom: Node) -> Node {
-    Node::Pair(Box::new(node), Box::new(atom))
-}
+#[derive(Clone, PartialEq, Debug)]
+pub struct Pair<T>(pub Box<T>, pub Box<T>);
```
Comment (hch12907) [inline src/node.rs:18]: `T` needs a `PartialEq` bound too, I think?

**Comment by zypeh** [comment]
@hch12907 what does it mean by `T` needs a `PartialEq` bound?

**Comment by hch12907** [comment]
`pub struct Pair<T: PartialEq>(pub Box<T>, pub Box<T>);`

**Pull request merged**